packages/compiler/core/checker.ts

```diff
@@ -3219,6 +3252,21 @@ export function createChecker(program: Program): Checker {
     return type;
   }
 
+  function resolveTypeReferenceString(reference: string): Type | undefined {
```

Comment:
Should we have this function in the checker or just move it up to the program(It is already in the program as `resolveTypeReference` calling out to this one)

Reply:
Wondering if it reallys belongs to the checker as it kinda calls out to a lot of things the checker never does otherwise(parser, binder) and it also has to have a different name as `resolveTypeReference` is already taken

Reply:
I think it would be OK to put this on Program.